src/librbd/ImageCtx.cc

```diff
@@ -59,7 +59,7 @@ class ThreadPoolSingleton : public ThreadPool {
                  "rbd_op_threads") {
     start();
   }
-  virtual ~ThreadPoolSingleton() {
+  ~ThreadPoolSingleton() {
```

Comment:
hi dillaman, thanks 

FYI here https://github.com/isocpp/CppCoreGuidelines/pull/737/file

CppCoreGuide said it's not prefer